### PR TITLE
issue: 770746 Add environment check before starting

### PR DESCRIPTION
**Pull request by igor-ivanov**
Current check services experimental_256B branch only
at the moment but can be extended later for different
purpose.
see https://github.com/Mellanox/libvma/pull/93 for details
this commit can be moved to experimental_256B branch after merge

### CODE REVIEW

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/107/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/108/ for details (Mellanox internal link).

File: contrib/jenkins_tests/globals.sh [hunk 1]
```diff
@@ -64,6 +64,18 @@ function do_github_status()
     "https://api.github.com/repos/$repo/statuses/${sha1}?access_token=$token"
 }
 
+function check_env()
+{
+	if [ ! $(git rev-parse --abbrev-ref HEAD | grep 'experimental_256B' >/dev/null 2>&1 || echo $?) ]; then
+		if [ $(ofed_info -s | grep 'MLNX_OFED_LINUX-3.2' >/dev/null 2>&1 || echo $?) ]; then
```
Comment (miked-mellanox) [inline contrib/jenkins_tests/globals.sh]: is there additional indication in mofed headers that libvma will not compile?
maybe it is better to grep mofed header file for support instead of rely on specific version?

Comment (igor-ivanov) [inline contrib/jenkins_tests/globals.sh]: I think that it is the simplest check at the moment. This branch is not stable and I do not think we need to use complex logic here.

Comment (miked-mellanox) [inline contrib/jenkins_tests/globals.sh]: :+1:

**Comment by OphirMunk** [comment]
if [ ! $(git rev-parse --abbrev-ref HEAD | grep 'experimental_256B' >/dev/null 2>&1 || echo $?) ]; then
	if [ $(ofed_info -s | grep 'MLNX_OFED_LINUX-3.2' >/dev/null 2>&1 || echo $?) ]; then

Instead of checking about being experimental_256B branch please consider checking about NOT being master branch

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/10/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/127/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/11/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/128/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
failed nodes do not have mofed 3.2. So it is a reason of failure. 
It is possible to return exit code as 0 but probably we need to return 1 in case environment is not suitable.
Are there any comments?

**Comment by miked-mellanox** [comment]
yep, skip with errcode 0 sounds only way to go here.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/12/ for details (Mellanox internal link).

File: contrib/jenkins_tests/globals.sh [hunk 2]
```diff
@@ -64,6 +64,18 @@ function do_github_status()
     "https://api.github.com/repos/$repo/statuses/${sha1}?access_token=$token"
 }
 
+function check_env()
+{
+	if [ $(git rev-parse --abbrev-ref HEAD | grep 'master' >/dev/null 2>&1 || echo $?) ]; then
+		if [ $(ofed_info -s | grep 'MLNX_OFED_LINUX-3.2' >/dev/null 2>&1 || echo $?) ]; then
+		    echo "environment [NOT OK]"
+		    exit 0
```
Comment (miked-mellanox) [inline contrib/jenkins_tests/globals.sh]: btw, maybe we should create .tap file here w/ tag SKIPPED to distinguish later setups where testing was done or skipped.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/129/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/13/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/130/ for details (Mellanox internal link).

File: contrib/jenkins_tests/globals.sh (hunk 2 above)
Comment (igor-ivanov) [inline contrib/jenkins_tests/globals.sh]: I thought about the same but decided to do this in separate PR. Actually I have realized that my current check is wrong. Usually PRs are based on user specific branch so the check is incorrect.

Comment (miked-mellanox) [inline contrib/jenkins_tests/globals.sh]: you can use jenkins built-in env variables available during jenkins run:

http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/130/injectedEnvVars/

Comment (igor-ivanov) [inline contrib/jenkins_tests/globals.sh]: @miked-mellanox It seems 'ghprbTargetBranch' can be used. Thank you for pointing to this.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/14/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/131/ for details (Mellanox internal link).

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-redmine-issue/24/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/141/ for details (Mellanox internal link).

**Comment by igor-ivanov** [comment]
@OphirMunk it is ready to be merged.
